tardis/visualization/tools/tests/test_sdec_plot.py

```diff
@@ -0,0 +1,295 @@
+"""Tests for SDEC Plots."""
+from tardis.base import run_tardis
+from tardis.io.config_reader import Configuration
+import pytest
+import pandas as pd
+import numpy as np
+from copy import deepcopy
+from tardis.visualization.tools.sdec_plot import SDECData, SDECPlotter
+import astropy.units as u
+import h5py
+
+
+@pytest.fixture(scope="module")
+def config_verysimple(atomic_dataset):
+    config = Configuration.from_yaml(
+        "tardis/io/tests/data/tardis_configv1_verysimple.yml"
+    )
+    return config
+
+
+@pytest.fixture(scope="module")
+def simulation_simple(config_verysimple, atomic_dataset):
+    """Instantiate SDEC plotter using a simple simulation model."""
+    # Setup simulation configuration using config_verysimple and
+    # override properties in such a way to make the simulation run faster
+    config_verysimple.montecarlo.iterations = 3
+    config_verysimple.montecarlo.no_of_packets = 4000
+    config_verysimple.montecarlo.last_no_of_packets = -1
+    config_verysimple.spectrum.virtual.virtual_packet_logging = True
+    config_verysimple.spectrum.num = 2000
+
+    atomic_data = deepcopy(atomic_dataset)  # TODO: why deepcopy?
+    sim = run_tardis(
+        config_verysimple,
+        atom_data=atomic_data,
+        show_convergence_plots=False,
+    )
+    return sim
+
+
+class TestSDECPlotter(object):
+    @classmethod
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, request):
+        cls = type(self)
+        cls.hdf_file = h5py.File("sdec_ref.h5", "a")
+        cls.save = False
+        cls.ref_data_path = ""
+        if request.config.getoption("--generate-reference"):
+            cls.save = True
+
+    @pytest.fixture(scope="class")
+    def plotter(self, simulation_simple):
+        return SDECPlotter.from_simulation(simulation_simple)
+
+    @pytest.mark.parametrize("species", [["Si II", "Ca II", "C", "Fe I-V"]])
+    def test_parse_species_list(self, request, plotter, species):
+        plotter._parse_species_list(species)
+        subgroup_name = request.node.callspec.id
```

Comment:
Does it return the combination of parameters as a string that also shows up on terminal when pytest runs different combinations?

Reply:
Yes, they can be seen if the user chooses to increase the verbosity. `request.node.callspec.id` returns the same. I am using those IDs to form different groups within the `hdf` file. Maybe we can discuss if this is an effective strategy in the coming meetings!